File: CS325/Misc/binarySearch.py
def binarySearch(list,val):
    if(len(list) > 0):
        mid = len(list)//2
        if(list[mid] == val):
            return True
        elif(val < list[mid]):
            start = 0
            end = mid
        else:
            start = mid + 1
            end = len(list)
            
        return binarySearch(list[start:end],val)
    return False

File: CS261/Search/binarySearch.c
int _binarySearch(TYPE * data, int size, TYPE val) 
{   
	int low  = 0;   
	int high = size;   

	while (low < high) 
	{     
		int mid = low + ((high - low) / 2); 
		
		if (data[mid] < val)
		{
			low  = mid + 1; 
		}
		else if(data[mid] > val)
		{		
			high = mid - 1;
		}
		else
		{
			return mid; //key found
		}
	}
	
	return -1; //key not found
}